internal/observability/middleware.go

```diff
@@ -152,14 +163,26 @@ func (m *Middleware) Handle(ctx context.Context, req *transport.Request, w trans
 	call := m.graph.begin(ctx, transport.Unary, _directionInbound, req)
 	defer m.handlePanicForCall(call, transport.Unary)
 
+	var requestSize int
+
+	if call.direction == _directionInbound {
```

Comment:
no need of this extra `call.direction == _directionInbound`

Handle is always the inbound middleware while Call is the outbound middleware